shared/health/src/main/java/io/pravega/shared/health/impl/HealthEndpointImpl.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.shared.health.impl;
+
+import io.pravega.shared.health.ContributorNotFoundException;
+import io.pravega.shared.health.ContributorRegistry;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.HealthContributor;
+import io.pravega.shared.health.HealthEndpoint;
+import io.pravega.shared.health.Status;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class HealthEndpointImpl implements HealthEndpoint {
+
+    private final ContributorRegistry registry;
+
+    /**
+     * Creates a new instance of the {@link HealthEndpointImpl} class.
+     * @param registry The {@link ContributorRegistry} to query.
+     */
+    HealthEndpointImpl(ContributorRegistry registry) {
```

Comment:
Tip: 1 use `@RequiredArgsConstructor` on the class, which will generate this for you.
TIp 2: Use `@NotNull` on the field in conjunction with Tip 1 above.